cpp/modmesh/view/RMainWindow.hpp

```diff
@@ -52,13 +54,15 @@ class RMainWindow
     }
 
     RPythonText * pytext() { return m_pytext; }
+    RPythonConsole * pyconsole() { return m_pyconsole.get(); }
     R3DWidget * viewer() { return m_viewer; }
 
 private:
 
     void setUp();
 
     RPythonText * m_pytext = nullptr;
+    std::shared_ptr<RPythonConsole> m_pyconsole = nullptr;
```

Comment:
A very good reason is needed to justify the use of shared pointer with Qt objects, which has its own ownership system.  Shared pointer does not work well with Qt pointers.  I do not see why you need the shared pointer here.

Reply:
I use shared_ptr because the pointer is shared. I think a raw pointer is also fine.

Reply:
You should use signal/slot for accessing other Qt objects to make the event loop and ownership manager happy.

Reply:
could you give some references?

Reply:
Googling the keywords in the sentence should be sufficient.